### PR TITLE
Tabulator 5

### LINKED ISSUE #19: Support SearchIterators as a data options
This would be great, would need to support lazy pagination.

Potentially could also support a function approach like ajax
e.g.
```python
anvil.server.call("get_page", page, page_size)
```

Would need to work out the practicals of sorting/filtering
Would also need to thinking about the return value from `getData()` - ideally the row object passed.

### PR DESCRIPTION
**Pull request by s-cork**
implement tabulator 5 with breaking change api

**Todo**:
- [x] add migration documentation
- [x] add documentation
- [x] support data being a SearchIterator (optional - probably for a future release)
- [x] clean up the example to be more readable - it's currently a bit of a mess
- [x] add docs for `tabulator.options["app_table"] = app_tables.my_table`

close #19

### CODE REVIEW

**Comment by s-cork** [comment]
@antletem, @steininge, @mjmare, @Starwort, @lewysigns, @samcarswell , @claresloggett 

As previous issue raisers/ contributors you may be interested in checking out these changes before I make them officially public.
It moves Anvil Tabulator to Tabulator 5.x

You can see the source code in this PR or by looking at the `tabulator-5` branch in github:
https://github.com/s-cork/Tabulator/tree/tabulator-5
The README/CHANGELOG highlight some of the new features/breaking changes.

The current `Development` version of the Tabulator Dependency points to this PR so you can also try it now by:
- switching the third party dependency `TGQCF3WT6FVL2EM2 ` to `Development`
- Recloning the code from the README clone links

I'll probably look to make a forum announcement and bump the version to `v2.0.0` next week at some point.
Please free to review any changes in this PR or add any requests 

Ta

**Comment by Starwort** [comment]
I can't tell you about the others, but I've largely moved away from anvil, myself; however, it seems like it would be worthwhile to include a specific migration guide? Both the README and the CHANGELOG seem rather focused on the new features, compared to what you've *changed* and what the response needs to be. (This is based on a brief skimming of both documents, and I may be wrong - but that's my initial impression)

**Comment by s-cork** [comment]
good point @Starwort I'll switch up the sections in the CHANGELOG - the second half (from line 200) talks about breaking changes - so I'll switch that to be the first half to be easier.

File: CHANGELOG.md [hunk 1]
```diff
@@ -369,12 +374,12 @@ self.tabulator.options = {
...
 
 The reason for this change is that something like, `pagination_size_selector`, can also take a value of `True`.
-That can't be expressed in the designer.
+This can't easily be expressed in the designer view of a property.
 
 
 ### row_selectable
 
-If you made use of the `row_selectable` column with checkboxes. That feature has been removed.
+If you made use of the `row_selectable` column with checkboxes. That feature has been removed as a propert.
```
Comment (antletem) [inline CHANGELOG.md:186]: Typo , propert -> property

**Comment by antletem** [comment]
> passed

To clarify , the imports to do not work if importing with the ```import tabulator.Tabulator``` naming convention. They now work with ```import Tabulator.Tabulator```

**Comment by s-cork** [comment]
Thanks @antletem 

Oh that's a bug. The package name here is correct so I must have done a bad push to the anvil remote. Thanks for reporting. 

I'll sort that in the next hour or so. Will post here when I fix it.

**Comment by s-cork** [comment]
@antletem that's fixed - sorry about that - I merged an autoupdate for pre-commit-cli - which the bumped the master version on anvil 🤦

**Comment by antletem** [comment]
> @antletem that's fixed - sorry about that - I merged an autoupdate for pre-commit-cli - which the bumped the master version on anvil 🤦

No Worries! that's why its not master yet. Thanks for fixing it so fast.

**Comment by s-cork** [comment]
@antletem I think I'm about ready to ship this
As the primary beta tester - apart from #21 - anything else to report?

**Comment by antletem** [comment]
> @antletem I think I'm about ready to ship this As the primary beta tester - apart from #21 - anything else to report?

Nope, It's working great!

**Comment by antletem** [comment]
> @antletem I think I'm about ready to ship this As the primary beta tester - apart from #21 - anything else to report?

Tangential to  ```get_selected_data()``` ,  I am setting  ```Tabulator.default_options["selectable"] = False``` for a class of tabulators and ``` Tabulator.default_options["selectable"] = True``` for a seperate class of tabulators. I've tried setting these default options to at runtime and before the classes are declared, but it seems to always set selectable to True. I am assuming that module is called first and sets the option for all future classes. Is that the case?

**Comment by s-cork** [comment]
That's expected, the default options are class level and only get created the first time a tabulator instance is created. A bit like theme or modules. 

Once you've set the default options and made a Tabulator instance, that's it. 

The remainder of the time you'll have to use the `self.tabulator.options`

**Comment by s-cork** [comment]
Merging as the development branch (since it already is)

**Comment by antletem** [comment]
> @antletem I've added support for that now. The default options, when changed, will be applied to any new Tabulator instances.

It looks like this feature was reverted back? I am on the development branch.

**Pull request merged**